test/test_public_bindings.py

```diff
@@ -272,6 +272,663 @@ def test_no_new_bindings(self):
         msg = f"torch._C had bindings that are not present in the allowlist:\n{difference}"
         self.assertTrue(torch_C_bindings.issubset(torch_C_allowlist_superset), msg)
 
+    def test_correct_module_names(self):
+        '''
+        An API is considered public, if  its  `__module__` starts with `torch.`
+        and there is no name in `__module__` or the object itself that starts with “_”.
+        Each public package should either:
+        (preferred) Define `__all__` and all non-module objects in there must have their `__module__` start with the current submodule's path.
```

Comment:
nit: add bullet here to differentiate the 2 cases from the two lines used to describe the first one.